File: languages/java/src/main/java/com/bloock/sdk/entity/identity/CredentialOfferCredential.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.IdentityEntities;

public class CredentialOfferCredential {
  private final String id;
  private final String description;

  public CredentialOfferCredential(String id, String description) {
    this.id = id;
    this.description = description;
  }

  public static CredentialOfferCredential fromProto(
      IdentityEntities.CredentialOfferBodyCredentials res) {
    return new CredentialOfferCredential(res.getId(), res.getDescription());
  }

  public IdentityEntities.CredentialOfferBodyCredentials toProto() {
    return IdentityEntities.CredentialOfferBodyCredentials.newBuilder()
        .setId(this.id)
        .setDescription(this.description)
        .build();
  }

  public String getId() {
    return id;
  }

  public String getDescription() {
    return description;
  }
}

File: languages/python/bloock/entity/identity/credential_offer_credential.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_pb2 as proto


class CredentialOfferCredential:
    def __init__(self, id: str, description: str) -> None:
        self.id = id
        self.description = description

    @staticmethod
    def from_proto(c: proto.CredentialOfferBodyCredentials) -> CredentialOfferCredential:
        return CredentialOfferCredential(
            id=c.id,
            description=c.description
        )

    def to_proto(self) -> proto.CredentialOfferBodyCredentials:
        return proto.CredentialOfferBodyCredentials(
            id=self.id,
            description=self.description
        )
